### PR TITLE
GH-431 federated repos not type specific

### LINKED ISSUE #431: federated repositories missing properties supported in local repositories
**Describe the bug**
A Federated repository should have the same set of properties supported as a local repository resource.

**Requirements for and issue**
```shell
terraform version
Terraform v1.1.7
on linux_amd64
+ provider registry.terraform.io/jfrog/artifactory v6.5.1

curl -s https://artifactory.com/artifactory/api/system/version | jq .version
"7.35.2"
```

```hcl2
resource "artifactory_federated_docker_repository" "my" {
  key             = "my"
  max_unique_tags = 10
  member {
    url     = "http://localhost"
    enabled = true
  }
}

Error: Unsupported argument
│ 
│   on federated.tf line 247, in resource "artifactory_federated_docker_repository" "my":
│  247:   max_unique_tags = 10
│ 
│ An argument named "max_unique_tags" is not expected here.
```
**Expected behavior**
The only difference between local and federated resources should be federation members. All other repository properties must be the same.

**Additional context**
Here is a full list of discrepancies between local and federated generated using [tfschema](https://github.com/minamijoyo/tfschema) tool

script:
```shell
types=$(tfschema resource list artifactory | grep federated | cut -d_ -f3)
for type in $types; do
    if [[ "${type}" == "docker" ]]; then
        continue
    fi
    echo $type
    diff <(tfschema resource show -format=json artifactory_local_${type}_repository | jq --sort-keys .attributes[].name) <(tfschema resource show -format=json artifactory_federated_${type}_repository | jq --sort-keys .attributes[].name)
    echo ""
done
echo docker_v1
diff <(tfschema resource show -format=json artifactory_local_docker_v1_repository | jq --sort-keys .attributes[].name) <(tfschema resource show -format=json artifactory_federated_docker_repository | jq --sort-keys .attributes[].name)
echo docker_v2
diff <(tfschema resource show -format=json artifactory_local_docker_v2_repository | jq --sort-keys .attributes[].name) <(tfschema resource show -format=json artifactory_federated_docker_repository | jq --sort-keys .attributes[].name)
```
output:
```
alpine
8d7
< "index_compression_formats"
12d10
< "primary_keypair_ref"

bower

cargo
1d0
< "anonymous_access"
9d7
< "index_compression_formats"

chef

cocoapods

composer

conan

conda

cran

debian
8d7
< "index_compression_formats"
12d10
< "primary_keypair_ref"
18,19d15
< "secondary_keypair_ref"
< "trivial_layout"

gems

generic

gitlfs

go

gradle
3d2
< "checksum_policy_type"
7,8d5
< "handle_releases"
< "handle_snapshots"
12d8
< "max_unique_snapshots"
20,21d15
< "snapshot_version_behavior"
< "suppress_pom_consistency_checks"

helm

ivy
3d2
< "checksum_policy_type"
7,8d5
< "handle_releases"
< "handle_snapshots"
12d8
< "max_unique_snapshots"
20,21d15
< "snapshot_version_behavior"
< "suppress_pom_consistency_checks"

maven
3d2
< "checksum_policy_type"
7,8d5
< "handle_releases"
< "handle_snapshots"
12d8
< "max_unique_snapshots"
20,21d15
< "snapshot_version_behavior"
< "suppress_pom_consistency_checks"

npm

nuget
6d5
< "force_nuget_authentication"
10d8
< "max_unique_snapshots"

opkg

puppet

pypi

rpm
3d2
< "calculate_yum_metadata"
6d4
< "enable_file_lists_indexing"
13d10
< "primary_keypair_ref"
19d15
< "secondary_keypair_ref"
21,22d16
< "yum_group_file_names"
< "yum_root_depth"

sbt
3d2
< "checksum_policy_type"
7,8d5
< "handle_releases"
< "handle_snapshots"
12d8
< "max_unique_snapshots"
20,21d15
< "snapshot_version_behavior"
< "suppress_pom_consistency_checks"

vagrant

docker_v1
1d0
< "api_version"
4d2
< "block_pushing_schema1"
11d8
< "max_unique_tags"
19d15
< "tag_retention"
docker_v2
1d0
< "api_version"
4d2
< "block_pushing_schema1"
11d8
< "max_unique_tags"
19d15
< "tag_retention"
```

### PR DESCRIPTION
**Pull request by alexhung**
Fixes #431 

Fix repo type specific attributes from local repo schema were not used for federated repo schemas.

Also fix the Artifactory auth token env var name in test scripts.

### CODE REVIEW

**Comment by alexhung** [comment]
@chb0github ☝🏼

File: pkg/artifactory/resource/repository/local/resource_artifactory_local_alpine_repository.go [hunk 1]
```diff
@@ -6,17 +6,16 @@ import (
 	"github.com/jfrog/terraform-provider-shared/util"
 )
 
-func ResourceArtifactoryLocalAlpineRepository() *schema.Resource {
-	const packageType = "alpine"
+var alpineLocalSchema = util.MergeSchema(BaseLocalRepoSchema, map[string]*schema.Schema{
```
Comment (chb0github) [inline pkg/artifactory/resource/repository/local/resource_artifactory_local_alpine_repository.go:9]: I think this whole function call should be multi lined

File: pkg/artifactory/resource/repository/local/resource_artifactory_local_cargo_repository.go [hunk 2]
```diff
@@ -6,17 +6,16 @@ import (
 	"github.com/jfrog/terraform-provider-shared/util"
 )
 
-func ResourceArtifactoryLocalCargoRepository() *schema.Resource {
-	const packageType = "cargo"
+var cargoLocalSchema = util.MergeSchema(BaseLocalRepoSchema, map[string]*schema.Schema{
```
Comment (chb0github) [inline pkg/artifactory/resource/repository/local/resource_artifactory_local_cargo_repository.go:9]: Multi lined

**Review by chb0github — CHANGES_REQUESTED** [review]
I think that when defining these schema it's time to go multi line. I'm getting confused. Otherwise looks good

**Review by danielmkn — APPROVED** [review]

**Review by danielmkn — APPROVED** [review]

**Review by chb0github — APPROVED** [review]

**Pull request merged**